src/geometry.rs

```diff
@@ -116,6 +116,24 @@ impl Ray {
         None
     }
 
+    pub fn box_intersection(&self, bbox: &BoundingBox, dist: &mut f32) -> bool {
+        let mut tmin = f32::NEG_INFINITY;
+        let mut tmax = f32::INFINITY;
+
+        let dir = 1.0 / self.vector;
```

Comment:
I think I've found the bug: We would expect `self.end - self.origin`, but `self.vector` is `self.origin - self.end`.

Reply:
Oh, in that case, the direction is backwards.
Weird, the test should have failed. I'll take a look at it too.